project-templates/node-json-theia/glsp-server/src/handler/delete-task-node-handler.ts

```diff
@@ -25,20 +26,18 @@ export class DeleteTaskNodeHandler implements OperationHandler {
     protected modelState: TaskListModelState;
 
     execute(operation: DeleteElementOperation): MaybePromise<void> {
-        operation.elementIds.forEach(elementId => this.deleteTask(elementId));
+        operation.elementIds.forEach(elementId => this.delete(elementId));
     }
 
-    protected deleteTask(elementId: string): void {
+    protected delete(elementId: string): void {
         const index = this.modelState.index;
-        let task = index.findTask(elementId);
-        if (!task) {
-            const taskNode = index.findParentElement(elementId, toTypeGuard(GNode));
-            if (taskNode) {
-                task = index.findTask(taskNode.id);
-            }
-        }
-        if (task) {
-            remove(this.modelState.taskList.tasks, task);
+        const node = index.findParentElement(elementId, toTypeGuard(GNode));
+        const elementIdToDelete = node ? node.id : elementId;
+        const elementToDelete = index.findTaskOrTransition(elementIdToDelete);
+        if (Task.is(elementToDelete)) {
+            remove(this.modelState.taskList.tasks, elementToDelete);
```

Comment:
Removing a task should also remove connected transitions.
Otherwise we end up with orphaned transitions in the tasklist source file.